docs/en/sql-reference/functions/array-functions.md

```diff
@@ -2118,6 +2118,113 @@ Result:
 └─────────────────────┘
 ```
 
+
+## randomSampleFromArray
+
+Returns a random sample from an input array. The number of elements in the sample is determined by the `sampleSize` argument. The function ensures that each element in the input array has an equal chance of being included in the sample.
+
+**Syntax**
+
+```sql
+randomSampleFromArray(arr, sampleSize)
+```
+
+**Arguments**
+
+- `arr` — The input array from which to sample elements, including nested arrays.
+- `sampleSize` — The number of elements to include in the random sample.
```

Comment:
Haven't looked at the code yet, but you want to specify the exact data types here. In particular, `sampleSize` could be made an unsigned integer. This would prevent the the case that it is negative (in which case there are multiple, not immediately intuitive options, e.g. 1. return the input array as is, 2. return an empty array, return NULL, 4. unspecified etc.)

Reply:
done

Reply:
Could we document also here, that sampleSize is an unsigned integer?

Reply:
I wrote "done", but then pushes the fix with some delay. This is probably the reason that you wasn't saw it. I hope now it is OK.